language/compiler/src/unit_tests/function_tests.rs

```diff
@@ -84,3 +84,23 @@ fn compile_script_with_large_frame() {
     let compiled_module_res = compile_module_string(&code);
     assert!(compiled_module_res.is_ok());
 }
+
+#[test]
+fn compile_module_with_script_visibility_functions() {
+    let code = String::from(
+        "
+        module Foobar {
+            public(script) foo() {
+                return;
+            }
+
+            public(script) bar() {
+                Self.foo();
+                return;
+            }
+        }
+        ",
+    );
```

Comment:
I'm surprised we still have these sorts of tests still around, as they are incredibly painful to update. 

You can keep it if you want, but I would add an ir-testuite functional test too

Reply:
Actually maybe just wait for the downstream PR for those tests :)